apps/passport-client/src/appHooks.ts

```diff
@@ -20,24 +21,40 @@ import {
 import { loadUsingLaserScanner } from "./localstorage";
 import { AppError, AppState } from "./state";
 import { useSelector } from "./subscribe";
-import { hasSetupPassword } from "./user";
+import { findUserIdentityPCD, hasSetupPassword } from "./user";
 import { getLastValidVerifyUrl, maybeRedirect } from "./util";
 
-export function usePCDCollection(): PCDCollection {
+/**
+ * Returns the user's PCDCollection, and also a wrapped version of the same
+ * collection.  The wrapped version is guaranteed to change (be replaced with
+ * a new object wrapping the same PCDCollection) each time the contents of
+ * PCDCollection changes.  It can be used in dependency lists for React code
+ * which needs to recalculate any time the contents of the PCDCollection change,
+ * even if the PCDCollection itself is not replaced.
+ *
+ * This wrapper-based approach may re-render unnecessarily if PCDCollection's
+ * change is a nop (such as re-issuing the same tickets), but is much cheaper
+ * than analyzing and hashing the full PCDCollection contents.
+ */
+export function usePCDCollectionWithWrapper(): [
+  PCDCollection,
+  Wrapper<PCDCollection>
+] {
```

Comment:
If you're returning a wrapped version, I think there is not a real reason to also return the unwrapped version. Could you explain your reasoning here a little bit?

Reply:
I was thinking of minimizing change, and similarity to the old pattern, though you're right that it seems odd given the cost of unwrapping is near-zero.

Probably the real question is whether to keep supporting the unwrapped `usePCDCollection`.  It seems valid for cases which don't have dependency lists to deal with, but also easy to use unsafely.